src/Gateways/GpApiGateway.php

```diff
@@ -265,10 +266,10 @@ public function woocommerce_globalpayments_gpapi_settings( $settings ) {
 		if ( ! wc_string_to_bool( $settings['enabled'] ) ) {
 			return $settings;
 		}
-		if ( empty( $settings['merchant_contact_url'] ) || 50 < strlen( $settings['merchant_contact_url'] ) ) {
+		if ( empty( $settings['merchant_contact_url'] ) || 256 < strlen( $settings['merchant_contact_url'] ) ) {
 			add_action( 'admin_notices', function () {
 				echo '<div id="message" class="notice notice-error is-dismissible"><p><strong>' .
-				     __( 'Please provide a Contact Url (maxLength: 50). Gateway not enabled.', 'globalpayments-gateway-provider-for-woocommerce' ) . '</strong></p></div>';
+				     __( 'Please provide a Contact Url (maxLength: 256). Gateway not enabled.', 'globalpayments-gateway-provider-for-woocommerce' ) . '</strong></p></div>';
 			} );
```

Comment:
This if can be removed by adding the maxlength  custom attribute to the `merchant_contact_url` setting.

Reply:
The gateway should no longer be disabled if invalid data introduced for App Id, App Key. The data should be saved only if valid, when gateway is enabled. I think we need a different hook to achieve this.